### ISSUE TITLE
Node modules

### ISSUE THREAD

**Issue by joshbenaron** [rogchap/v8go#148]
Hey,
Does this work with node modules. If so how does it work?

**Comment by matthewmueller** [comment]
Yes if the node modules don't use node. I'd recommend bundling with https://esbuild.github.io/ then passing the bundle into gov8,.

**Comment by mars** [comment]
> Yes if the node modules don't use node.

I think this means things like `require` aren't available, because it's a higher-level Node API, not V8.

So, many modules will work, but if a module uses Node's APIs, they will fail at runtime. For example:

```
ReferenceError: require is not defined
```

I have not been able to find a work-around. So, I guess I'll have to pull Node itself into this project, run it as a subprocess, and implement a bridge API (perhaps using [go-plugin](https://github.com/hashicorp/go-plugin)).

**Comment by goenning** [comment]
I second what @matthewmueller suggested, that's what I use to run most npm packages on v8go

**Comment by timo-klarshift** [comment]
As mentioned I also recommend esbuild for this. Just dont use stuff from node like `fs` or `crypto` etc and esbuild will bundle everything to a single file you can load.

**Comment by sb-bilal-dev-ev** [comment]
Does single file builds with rollup work too?

I receive this error when trying to call an exported function bundle.js

Let's say I have deguasser.js (bundle file). And I have the following `.go` code

```golang
func callDegausser(node string, options string) {
	ctx := v8.NewContext()
	
	importDegausser := `var degausser = require('./degausser.js');`
	consoleLog := `console.log('WORKING!'); `
	callDegausser := importDegausser + consoleLog + `degausser('` + "" + `','` + "" + `')`
	
	// val, _ := ctx.RunScript(importDegausser + callDegausser, "degausser.js") // executes a script on the global context
	val, err := ctx.RunScript(callDegausser, "degausser.js") // executes a script on the global context

	log.Println("value: ", val, err)
}

```

I'm receving `ReferenceError: require is not defined` error

**Comment by mars** [comment]
@sb-bilal-dev-ev as I [commented](https://github.com/rogchap/v8go/issues/148#issuecomment-903031001) above, 

`require` is a Node API. It's not part of V8. It won't work in this context.

**Comment by thefuga** [comment]
I'm running into a similar issue.

I've bundled a small script that uses external modules with esbuild, but when importing packages (either through require or import keyword), I always get the `SyntaxError: Unexpected identifier` or similar syntax errors, with no further details.

Is there any way to make this more verbose so I can understand what's going on?
